app/commands/track/update_build_status.rb

```diff
@@ -0,0 +1,359 @@
+class Track::UpdateBuildStatus
+  include Mandate
+
+  initialize_with :track
+
+  def call
+    track.build_status = build_status
+  end
+
+  private
+  def build_status
+    {
+      health:,
+      volunteers:,
+      students:,
+      submissions:,
+      mentor_discussions:,
+      syllabus:,
+      practice_exercises:,
+      test_runner:,
+      representer:,
+      analyzer:
+    }
+  end
+
+  def health
+    num_exemplar = component_health_statuses[:exemplar].to_i
+    num_healthy = component_health_statuses[:healthy].to_i
+
+    return :exemplar if num_exemplar == 5
+    return :healthy if num_exemplar + num_healthy == 5
+
+    :needs_attention
+  end
+
+  memoize
+  def component_health_statuses = [
+    analyzer_health,
+    test_runner_health,
+    representer_health,
+    syllabus_health,
+    practice_exercises_health
+  ].tally
+
+  def volunteers
+    volunteer_user_ids = User::ReputationPeriod.where(
+      period: :forever,
+      about: :track,
+      track_id: track.id
+    ).select(:user_id)
+    volunteers = User.with_attached_avatar.where(id: volunteer_user_ids).order(reputation: :desc).take(12)
```

Comment:
A lot of users have `avatar_url` field set, rather than having had an avatar uploaded (e.g. if we took the data from the GH oauth). So we might want some sort of OR here.

Reply:
OK, so from our chat on Slack I understand this just does `include(...)` effectively.

What I was trying to suggest was that if we only want to show faces here, then we'd want to filter to get records either with an attached_avatar, or with `avatar_url` being a proper value. I can't remember how to do that efficiently though.